marker_maze.py

```diff
@@ -0,0 +1,242 @@
+        self.image_centre_y = image_height / 2.0
+        self.drive = drive
+        self.screen = screen
+        self.stream = picamera.array.PiRGBArray(camera)
+        self.event = threading.Event()
+        self.terminated = False
+        # Why the one second sleep?
+        #create small cust dictionary
+        self.small_dict = aruco.Dictionary_create(6, 3)
+        self.last_t_error = 0
+        self.TURN_P = 0.6
+        self.TURN_D = 0.3
+        self.STRAIGHT_SPEED = 0.5
+        self.STEERING_OFFSET = 0.0  #more positive make it turn left
+        self.CROP_WIDTH = 320
+        self.i = 0
+        self.TIMEOUT = 30.0
+        self.START_TIME = time.clock()
+        self.END_TIME = self.START_TIME + self.TIMEOUT
+        self.found = False
+        self.turn_number = 0
+        self.TURN_TARGET = 5
+        self.TURN_WIDTH = [30, 35, 35, 30, 35, 35]
+        self.NINTY_TURN = 0.8  #0.8 works if going slowly
+        self.SETTLE_TIME = 0.05
+        self.TURN_TIME = 0.04
+        self.MAX_TURN_SPEED = 0.25
+        self.loop_start_time=0
+        self.marker_to_track=0
+        self.BRAKING_FORCE = 0.1
+        self.BRAKE_TIME = 0.05
+        self.finished = False
+        logger.info("setup complete, looking")
+        time.sleep(1)
+        self.start()
+
+    def run(self):
+        # This method runs in a separate thread
+        while not self.terminated:
+            # Wait for an image to be written to the stream
+            if self.event.wait(1):
+                try:
+                    # Read the image and do some processing on it
+                    self.stream.seek(0)
+                    self.process_image(self.stream.array, self.screen)
+                finally:
+                    # Reset the stream and event
+                    self.stream.seek(0)
+                    self.stream.truncate()
+                    self.event.clear()
+
+    def turn_right(self):
+        self.drive.move(self.NINTY_TURN, 0)
+        time.sleep(self.TURN_TIME)
+        self.drive.move(0,0)
+        time.sleep(self.SETTLE_TIME)
+                
+    def turn_left(self):
+        self.drive.move(-self.NINTY_TURN, 0)
+        time.sleep(self.TURN_TIME)
+        self.drive.move(0,0)
+        time.sleep(self.SETTLE_TIME)
+
+    def brake(self):
+        self.drive.move(0,-self.BRAKING_FORCE)
+        time.sleep(self.BRAKE_TIME)
+        self.drive.move(0,0)
+    
+    def process_image(self, image, screen):
+        screen = pygame.display.get_surface()
+        if self.turn_number >= self.TURN_TARGET:
+           logger.info("finished!")
+           self.finished = True
+        frame = image[30:190, (self.image_centre_x - self.CROP_WIDTH/2):(self.image_centre_x + self.CROP_WIDTH/2)]
+        # Our operations on the frame come here
+        gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+        parameters =  aruco.DetectorParameters_create()
+        #print(parameters)
+        '''    detectMarkers(...)
```

Comment:
Why the inline docstring?

Reply:
as from create_aruco_dict.py, it can be removed